drivers/tee/tee_core.c

```diff
@@ -111,6 +121,209 @@ static int tee_release(struct inode *inode, struct file *filp)
 	return 0;
 }
 
+static const char *tee_session_get_application_id(void)
+{
+	return NULL;
+}
+
+static void tee_session_free_application_id(const char *app_id)
+{
+}
+
+int tee_session_verify_perm(u32 connection_method,
+			    const u8 connection_data[TEE_IOCTL_UUID_LEN])
+{
+	gid_t ns_grp = (gid_t)-1;
+	kgid_t grp = INVALID_GID;
+
+	/* Verify group memberships for logins including group */
+	if ((connection_method == TEE_IOCTL_LOGIN_GROUP) ||
+	    (connection_method == TEE_IOCTL_LOGIN_GROUP_APPLICATION)) {
+		memcpy(&ns_grp, connection_data, sizeof(gid_t));
+		grp = make_kgid(current_user_ns(), ns_grp);
+
+		if (!gid_valid(grp) || !in_egroup_p(grp))
+			return -EPERM;
+	}
+
+	return 0;
+}
+EXPORT_SYMBOL_GPL(tee_session_verify_perm);
+
+static int uuid_v5(uuid_t *uuid, const uuid_t *ns, const void *name,
+		   size_t size)
+{
+	unsigned char hash[SHA1_DIGEST_SIZE];
+	struct crypto_shash *shash = NULL;
+	struct shash_desc *desc = NULL;
+	int rc;
+
+	shash = crypto_alloc_shash("sha1", 0, 0);
```

Comment:
Any particular reason why we're using SHA-1?

Reply:
UUIDv5 is SHA1 based. Much better than UUIDv3 which is MD5 based.

Reply:
I see. A comment explaining that we're hashing for UUIDv5 as described in RFCxxxx would be useful.

Reply:
method is called as uuid_v5 :) ? I can add comment about RFC too there.